### ISSUE TITLE
Attic output contains later introduced tag

### ISSUE THREAD

**Issue by tyrasd** [drolbr/Overpass-API#241]
The following query

    [out:xml][timeout:25][date:"2012-10-01T00:00:01"];
    way(30713396);
    out tags;

which basically requests version 7 of way 30713396, returns the tag `tunnel` with a very suspicious value (see later). The way only recently got this tag added, see http://osmlab.github.io/osm-deep-history/#/way/30713396. Looking at the returned content, the value of the tag appears to contain quite strange data: Running ` curl http://overpass-api.de/api/interpreter?data=%5Bout%3Axml%5D%5Btimeout%3A25%5D%5Bdate%3A%222012-10-01T00%3A00%3A01%22%5D%3B%0Away%2830713396%29%3B%0Aout%20tags%3B | hexdump -C` produces:

    […]
    00000150  67 20 6b 3d 22 74 75 6e  6e 65 6c 22 20 76 3d 22  |g k="tunnel" v="|
    00000160  ff 20 22 2f 3e 0a 20 20  3c 2f 77 61 79 3e 0a 0a  |. "/>.  </way>..|
    00000170  3c 2f 6f 73 6d 3e 0a                              |</osm>.|

i.e. the string starts with `0xff` which doesn't seem to be valid utf-8, is it?

**Comment by mmd-osm** [comment]
Imho 0xFF is being used as an internal identifier for a void (empty) tag
value (see
https://github.com/drolbr/Overpass-API/blob/a0b334e8fcbee40f05cf74c0a744584c18f4d711/src/overpass_api/core/type_tags.h#L122
). I really wonder why this made it to the query result. Looks like a bug
to me.

**Comment by mmd-osm** [comment]
Imho `0xff` is being used as an internal identifier for a void (empty) tag value (see [here](https://github.com/drolbr/Overpass-API/blob/a0b334e8fcbee40f05cf74c0a744584c18f4d711/src/overpass_api/core/type_tags.h#L122)). However, tag "tunnel" really never existed until version 24 and it's a bit strange that the tag contains `0xff 0x20` as value.

Looks like a bug to me. Unfortunately, attic issues are the most difficult ones to analyze, so that's another one for Roland...

**Comment by mmd-osm** [comment]
More candidates:

```C
[out:xml][timeout:25][date:"2012-10-01T00:00:01"];
way(60966427);
out tags;
```

**Comment by mmd-osm** [comment]
Further analysis:

`dump_file: --attic-way-tags-local | grep $'[\xff][\x20]' | cut -f 2 | sort | uniq -c`

-> A total of **141497 ways** is affected by this issue, see: http://dev.overpass-api.de/issue241/way_candidates.txt

**Comment by drolbr** [comment]
Fixed in 919f06ca84965794ee47d8fbb3287c2ed2d6d74a .

A rebuild of the database could be avoided if we told every using function to ignore tag with those values. But due to bug 276 it makes sense to rebuild to database anyway.
